moveit_ros/planning_interface/test/move_group_interface_cpp_test.cpp

```diff
@@ -0,0 +1,463 @@
+
+static const std::string PLANNING_GROUP = "panda_arm";
+constexpr double PLANNING_TIME_S = 10;
+
+class MoveGroupTestFixture : public ::testing::Test
+{
+public:
+  void SetUp() override
+  {
+    nh_ = ros::NodeHandle("/move_group_interface_cpp_test");
+    move_group_ = std::make_shared<moveit::planning_interface::MoveGroupInterface>(PLANNING_GROUP);
+
+    // set velocity and acceleration scaling factors (full speed)
+    move_group_->setMaxVelocityScalingFactor(1.0);
+    move_group_->setMaxAccelerationScalingFactor(1.0);
+  }
+
+  // Helper function to move to a start position
+  void MoveToStart(const geometry_msgs::Pose& pose)
+  {
+    // get the joint model group
+    const robot_state::JointModelGroup* joint_model_group =
+        move_group_->getCurrentState()->getJointModelGroup(PLANNING_GROUP);
+
+    robot_state::RobotState start_state(*move_group_->getCurrentState());
+    start_state.setFromIK(joint_model_group, pose);
+    move_group_->setStartState(start_state);
+
+    // plan and move
+    ASSERT_TRUE(move_group_->setJointValueTarget(start_state));
+    moveit::planning_interface::MoveGroupInterface::Plan my_plan;
+    ASSERT_EQ(move_group_->plan(my_plan), moveit::planning_interface::MoveItErrorCode::SUCCESS);
+    ASSERT_EQ(move_group_->move(), moveit::planning_interface::MoveItErrorCode::SUCCESS);
+  }
+
+protected:
+  ros::NodeHandle nh_;
+  moveit::planning_interface::MoveGroupInterfacePtr move_group_;
+  moveit::planning_interface::PlanningSceneInterface planning_scene_interface_;
+};
+
+TEST_F(MoveGroupTestFixture, StartingConditionsTest)
+{
+  // test that setting the planning time works
+  move_group_->setPlanningTime(PLANNING_TIME_S);
+  EXPECT_EQ(move_group_->getPlanningTime(), PLANNING_TIME_S);
+
+  // test that the world and panda robot is initialized the way we expect by default
+  EXPECT_EQ(move_group_->getNodeHandle().getNamespace(), "/");
+  EXPECT_EQ(move_group_->getEndEffectorLink(), "panda_link8");
+  EXPECT_EQ(move_group_->getEndEffector(), "");
+  EXPECT_EQ(move_group_->getPoseReferenceFrame(), "world");
+  EXPECT_EQ(move_group_->getName(), "panda_arm");
+  EXPECT_EQ(move_group_->getPlanningFrame(), "world");
+  EXPECT_EQ(move_group_->getVariableCount(), std::size_t(7));
+  EXPECT_EQ(move_group_->getDefaultPlannerId(), "");
+  EXPECT_EQ(move_group_->getPlanningTime(), 10.0);
+  EXPECT_EQ(move_group_->getGoalJointTolerance(), 0.0001);
+  EXPECT_EQ(move_group_->getGoalPositionTolerance(), 0.0001);
+  EXPECT_EQ(move_group_->getGoalOrientationTolerance(), 0.001);
+
+  const std::vector<std::string> NAMED_TARGETS = { "ready", "extended" };
+  std::vector<std::string> named_targets = move_group_->getNamedTargets();
+  ASSERT_EQ(named_targets.size(), NAMED_TARGETS.size());
+  for (size_t i = 0; i < named_targets.size(); ++i)
+    EXPECT_EQ(named_targets[i], NAMED_TARGETS[i]) << "index: " << i;
+
+  const std::vector<std::string> JOINT_MODEL_GROUP_NAMES = { "hand", "panda_arm", "panda_arm_hand" };
```

Comment:
All these tests follow the same scheme. Please factor that out into a separate function.

Reply:
done